### PR TITLE
a word count application in a aws lambda.

### PR DESCRIPTION
**Pull request by infinite-Joy**
*Issue #, if available:*

*Description of changes:*

This is a simple word count example to help users understand how to use rust lambda capabilities in a function of their own.

By submitting this pull request

- [x] I confirm that my contribution is made under the terms of the Apache 2.0 license.
- [x] I confirm that I've made a best effort attempt to update all relevant documentation.

### CODE REVIEW

File: lambda-runtime/examples/word-count/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,58 @@
+#[macro_use]
```
Comment (sapessi) [inline lambda-runtime/examples/word-count/src/main.rs]: Perhaps we can switch to 2018 edition and remove all of the `extern` and `#[macro_use]`.

File: lambda-runtime/examples/word-count/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+fn main() -> Result<(), Box<dyn Error>> {
+    simple_logger::init_with_level(log::Level::Debug).unwrap();
+    lambda!(my_handler);
+
+    Ok(())
+}
+
+fn my_handler(event: CustomEvent, ctx: Context) -> Result<CustomOutput, HandlerError> {
```
Comment (sapessi) [inline lambda-runtime/examples/word-count/src/main.rs]: You don't need a `CustomOutput` here, you should be able to return the `HashMap` directly.

File: lambda-runtime/examples/word-count/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    if event.string == "" {
+        error!("Empty name in request {}", ctx.aws_request_id);
```
Comment (sapessi) [inline lambda-runtime/examples/word-count/src/main.rs]: Is it a name or a message/text you expect?

File: lambda-runtime/examples/word-count/src/main.rs (hunk 1 above)
Comment (infinite-Joy) [reply lambda-runtime/examples/word-count/src/main.rs]: sure. will test this and update the PR.

File: lambda-runtime/examples/word-count/src/main.rs (hunk 3 above)
Comment (infinite-Joy) [reply lambda-runtime/examples/word-count/src/main.rs]: Thanks for the catch. will update.

**Comment by infinite-Joy** [comment]
@sapessi I have made the changes. Thanks for the review. Please have a look.

**Comment by coltonweaver** [comment]
Because this PR is very outdated and hasn't been touched in almost 2 years, I'm going to close it out. If you wish to revive it, feel free to do so. Thanks!

**Pull request closed without merging**